packages/headless/src/controllers/commerce/search/did-you-mean/headless-did-you-mean.ts

```diff
@@ -39,6 +49,13 @@ export function buildDidYouMean(engine: CommerceEngine): DidYouMean {
   return {
     ...controller,
 
+    applyCorrection() {
```

Comment:
Now that I look at it, I don't think this is going to work.

Dispatching `applyCorrection` will set `queryCorrection.correctedQuery` to the desired value. Up to this point this is fine.

However, when we dispatch `executeSearch`, the action will use the `querySelector` to ge the current query... but this will be the old query.

Then, while the action is pending, the did you mean slice will clear the queryCorrection from the state and reset `wasCorrectedTo` to `''`... this is probably fine.

What I believe is missing, however, is that before we dispatch `executeSearch`, we should dispatch `updateQuery`. As a matter of fact, I'm not even sure it's worth updating `wasCorrectedTo` in the state at all: it'll get reset anyways.

Reply:
If you agree with what I'm saying, this changes things considerably: we don't need a new action (neither do we need an action loader, obviously), and everything is much simpler.

We just dispatch updateQuery, and then executeSearch.

Reply:
Ah I found how it works in search. There is a case that is changing the .q for applyDidYouMeanCorrection. 😆 . I thought we changed the wasCorrectedTo for it to say "query was corrected to ...". Turns out we change it for no reason ! I will change it to how you said.

```
export const queryReducer = createReducer(getQueryInitialState(), (builder) =>
  builder
    .addCase(updateQuery, (state, action) => ({...state, ...action.payload}))
    .addCase(applyDidYouMeanCorrection, (state, action) => {
      state.q = action.payload;
    })
```